### DISCUSSION TITLE
Ability to queue commands?

### DISCUSSION THREAD

**Discussion by andreincu** [lllyasviel/stable-diffusion-webui-forge#1686] [Q&A]
Discovered this Agent scheduler, but it's not working. Any alternatives?
https://github.com/ArtVentureX/sd-webui-agent-scheduler

**Comment by LeDXIII** [comment]
https://github.com/supersonic13/sd-webui-agent-scheduler

**Comment by Tablaski** [comment]
Very interested as well, installed the supersonic13 url extension

I can see all the buttons in forge but nothing happens when clicking enqueue
I don't even find errors in the forge terminal output

**Reply by andreincu** [reply]
I've discovered that using Plot X/Y/Z is sufficient most of the time.

Alternative would be to use comfy ui, which allows this by design.

**Reply by Tablaski** [reply]
There's also a possibility to import a prompt list / file 
But that's not the same as proper queue management

About Comfy UI, I don't like it for flux, Forge is much faster, I'd rather not come back to Comfy unless there's really no other option

**Reply by supersonic13** [reply]
Hello. I made small updates the fork today. Could you update it on your system and try again ? The enqueue button should work straightaway with whatever checkpoint is selected in the top bar.

**Reply by Tablaski** [reply]
I have just updated using the extension tab menu in Forge but it still doesn't enqueue anything :-/
What info can I bring to help ?

**Reply by supersonic13** [reply]
hi if you have any console logs, it'll be useful.  i'll also try to do a clean install and try it.

**Reply by supersonic13** [reply]
https://github.com/ArtVentureX/sd-webui-agent-scheduler/issues/256#issuecomment-2397014565
Not sure if its related, but another commentator indicated "Okay, it works, but only after I refresh the tab (Chrome, Windows 10). ..."

**Reply by Tablaski** [reply]
Sorry, indeed it works after refreshing the UI after its initial load. Maybe you can fix that later but I'm very happy.
API seems to work as well. 

Thank you so much, great work !!!

**Comment by Tablaski** [comment]
Some feedback for you @supersonic13 after using agent scheduler, to improve your extension. But first I want to say your work is amazing and I love it.

- It's very difficult to edit prompts in the tab because the field is a single line one instead of a paragraph (or a field that could be extended) 
- Probably same issue with negative prompt, haven't used it that much
- Rearranging the order of the queue items is difficult as well because it is relying on drag & drop only. It's ok when you have a few items in the queue, but if you have many it's a hassle, and if the items are not on page 1 then it's simply impossible
- It would be great to have queue first / queue last buttons if the main generation time, like in ComfyUI. And/or actions in the queue itself ?
- I've found out using the queue often make the Generate button hide the Interrupt and Skip buttons after a while ; not sure yet how to reproduce this though, or even if it's directly linked to the extension but it does happen. Perhaps these actions could be in the queue actions as well, or just in the queue actions if it's too difficult to fix the main tab buttons issue.
- It can happen to make mistakes when queueing and realizing it later, like wrong number of steps or wrong CFG. That means having to delete everything in the queue / try to queue them again by getting the prompt for the field, etc. Difficult & daunting task. It would be great to be able to batch apply settings such as CFG, batch size, steps, etc to all queue items, and/or even better to selected ones in the queue
- I'm not sure what is the purpose of the drag & drop zone for pictures at the right of the queue ? 
- Not a big requirement at all, but could be useful to know where we're at in the batch itself (like processing picture 5 of 10 in the batch of 10), if that's easy to implement maybe

I haven't tried the API yet but i'm on my to-do list

**Reply by supersonic13** [reply]
@Tablaski thanks ! but I cannot take credit for the extension. The original extension was developed by "ArtVentureX" and it is a brilliant one. I modified a few files/lines of code so it worked with the new Forge.
I am, with my limited knowledge, trying to create a simpler ComfyUI like extension to queue, and would be heavily inspired by the original agent scheduler - and will take your comments into account while developing that.

**Reply by Tablaski** [reply]
That is still very valuable for us :-) Thanks for working on this

**Reply by AIKS68** [reply]
> @Tablaski thanks ! but I cannot take credit for the extension. The original extension was developed by "ArtVentureX" ...
... and will take your comments into account while developing that.

I'm glad to have something that works! I also need to do a reload to get it working, but that isn't a big issue.
